src/account/AccountItem.js

```diff
@@ -88,36 +97,71 @@ export default function AccountItem(props: Props): Node {
   // `realmName` will be the real realm name, not the fallback.
   // TODO(#5005) look for server data even when this item's account is not
   //   the active one.
-  const realmName =
-    isActiveAccount && activeAccountState != null && getHaveServerData(activeAccountState)
-      ? getRealmName(activeAccountState)
-      : '(unknown organization name)';
+  let realmName = '(unknown organization name)';
+  let expiryWarning = null;
+  if (isActiveAccount && activeAccountState != null && getHaveServerData(activeAccountState)) {
+    realmName = getRealmName(activeAccountState);
+    expiryWarning = pushNotificationsEnabledEndTimestampWarning(activeAccountState, dateNow);
+  }
   const singleNotifProblem = chooseNotifProblemForShortText({
     report: notificationReport,
     ignoreServerHasNotEnabled: silenceServerPushSetupWarnings,
   });
 
   const handlePressNotificationWarning = React.useCallback(() => {
-    if (singleNotifProblem == null) {
+    if (expiryWarning == null && singleNotifProblem == null) {
+      logging.warn('AccountItem: Notification warning pressed with nothing to show');
+      return;
+    }
+
+    if (expiryWarning != null) {
```

Comment:
Can there be both an expiry warning and a single-notif-problem? In that case we'd probably want to prioritize the active problem, right?